plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -294,15 +273,10 @@ func NewNodePreferAvoidPodsPriority(controllerLister algorithm.ControllerLister,
 	return nodePreferAvoid.CalculateNodePreferAvoidPodsPriority
 }
 
-func (npa *NodePreferAvoidPod) CalculateNodePreferAvoidPodsPriority(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
-	nodes, err := nodeLister.List()
-	if err != nil {
-		return nil, err
-	}
-
+func (npa *NodePreferAvoidPod) CalculateNodePreferAvoidPodsPriority(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.NodeInfo, nodes []*api.Node) (schedulerapi.HostPriorityList, error) {
 	// TODO: Once we have ownerReference fully implemented, use it to find controller for the pod.
-	rcs, err := npa.controllerLister.GetPodControllers(pod)
-	rss, err := npa.replicaSetLister.GetPodReplicaSets(pod)
+	rcs, _ := npa.controllerLister.GetPodControllers(pod)
```

Comment:
This was also ignored before (but not explicitly).

Anyway - error from GetPodControllers() in most cases mean that there is no controller for a give pod (which basically is not even a real error). The interface of ReplicationControllerToStoreLister is misleading in my opinion and once OwnerRef is finished we will be able to get rid of that hopefullly. (I already have a different PR in my local branch that is changing it, so that's mostly temporary anyway).